adafruit_logging.py

```diff
@@ -101,63 +102,80 @@ def level_for(value: int) -> str:
 
-    def format(self, log_level: int, message: str) -> str:
+    def __init__(self, level=NOTSET):  # pylint: disable=undefined-variable
+        self.level = level
+        """Level of the handler; this is currently unused, and
+        only the level of the logger is used"""
+
+    def _format(self, log_level: int, message: str) -> str:
         """Generate a timestamped message.
 
         :param int log_level: the logging level
         :param str message: the message to log
 
         """
         return "{0:<0.3f}: {1} - {2}".format(
-            time.monotonic(), level_for(log_level), message
+            time.monotonic(), _level_for(log_level), message
         )
 
-    def emit(self, log_level: int, message: str):
+    def _emit(self, log_level: int, message: str):
         """Send a message where it should go.
         Placeholder for subclass implementations.
         """
         raise NotImplementedError()
 
 
-class PrintHandler(LoggingHandler):
-    """Send logging messages to the console by using print."""
+#  pylint: disable=too-few-public-methods
+class StreamHandler(Handler):
+    """Send logging messages to a stream, `sys.stderr` (typically
+    the serial console) by default.
+
+    :param stream: The stream to log to, default is `sys.stderr`
+    """
+
+    def __init__(self, stream=None):
+        super().__init__()
+        if stream is None:
+            stream = sys.stderr
+        self.stream = stream
+        """The stream to log to"""
 
-    def emit(self, log_level: int, message: str):
+    def _emit(self, log_level: int, message: str):
         """Send a message to the console.
 
         :param int log_level: the logging level
         :param str message: the message to log
 
         """
-        print(self.format(log_level, message))
+        print(self._format(log_level, message))
 
 
 # The level module-global variables get created when loaded
 # pylint:disable=undefined-variable
 
 logger_cache = {}
-null_logger = None
+
+
+def _addLogger(logger_name: str):
+    """Adds the logger if it doesn't already exist"""
+    if logger_name not in logger_cache:
+        logger_cache[logger_name] = Logger(logger_name)
+
 
 # pylint:disable=global-statement
 def getLogger(logger_name: str) -> "Logger":
-    """Create or retrieve a logger by name.
+    """Create or retrieve a logger by name; only retrieves loggers
+    made using this function
 
-    :param str logger_name: The name of the `Logger` to create/retrieve. `None`
-                            will cause the `NullLogger` instance to be returned.
+    :param str logger_name: The name of the `Logger` to create/retrieve.
 
     """
-    global null_logger
-    if not logger_name or logger_name == "":
-        if not null_logger:
-            null_logger = NullLogger()
-        return null_logger
-
-    if logger_name not in logger_cache:
-        logger_cache[logger_name] = Logger()
-    return logger_cache[logger_name]
+    _addLogger(logger_name)
```

Comment:
> `getLogger()` now returns None if the requested logger isn't found, as opposed to creating one

It looks like the _addLogger() call here is still creating the Logger if it wasn't already in the cache I think. Is that the intended behavior?

Reply:
Yes, that is the intended behavior, I should update the comment 😅 I could also switch the relevant `dict.get(log_name, None)` back to just dict[log_name] since it shouldn't ever fail in this case.  Right?

Reply:
Yeah, it does look like it could use the subscript access instead of dict.get() function. I don't have a huge preference of one over the other, but you can change it back to subscript if you like.